include/i2c.h

```diff
@@ -466,6 +511,8 @@ static inline int i2c_burst_write(struct device *dev, u16_t dev_addr,
  * This routine reads the value of an 8-bit internal register of an I2C
  * device synchronously.
  *
+ * @deprecated Replace with i2c_write_read().
```

Comment:
The rational for deprecating it is that its companion `i2c_burst_write()` simply does not work, so that has to be deprecated.  I don't mind removing `@deprecated`, but would like to see additional opinions before doing that.

Reply:
I'd also prefer that `i2c_reg_read_byte` stays, it's quite handy